package helen.resources;

//option java_package = "com.example.tutorial";
//option java_outer_classname = "AddressBookProtos";
option optimize_for = SPEED; 

message WebResource {
	required uint32 id = 1;			// id of a resource
	required int32 status = 2;		// status to be used by filters

	optional string url = 3;		// URL
	optional uint32 time = 4;	 	// last check
	optional string mime_type = 5;		// resource MIME type
	optional string content = 6;		// content of a resource

	repeated string header_field = 10;	// response header fields
	repeated string extracted_url = 11;	// URLs extracted from the resource
	optional fixed32 ip_addr = 12;		// IPv4 address (result of DNS query)
}
